tests/acl/test_acl.py

```diff
@@ -446,239 +480,288 @@ def get_dst_ports(self, setup, direction):
+            )
+
+    def icmp_packet(self, setup, direction, ptfadapter, ip_version, src_ip=None, dst_ip=None):
         """Generate an ICMP packet for testing."""
-        return testutils.simple_icmp_packet(
-            eth_dst=setup["router_mac"],
-            eth_src=ptfadapter.dataplane.get_mac(0, 0),
-            ip_dst=self.get_dst_ip(direction),
-            ip_src=DEFAULT_SRC_IP,
-            icmp_type=8,
-            icmp_code=0,
-            ip_ttl=64,
-        )
-
-    def expected_mask_routed_packet(self, pkt):
+        src_ip = src_ip or DEFAULT_SRC_IP[ip_version]
+        dst_ip = dst_ip or self.get_dst_ip(direction, ip_version)
+
+        if ip_version == "ipv4":
+            return testutils.simple_icmp_packet(
+                eth_dst=setup["router_mac"],
+                eth_src=ptfadapter.dataplane.get_mac(0, 0),
+                ip_dst=dst_ip,
+                ip_src=src_ip,
+                icmp_type=8,
+                icmp_code=0,
+                ip_ttl=64,
+            )
+        else:
+            return testutils.simple_icmpv6_packet(
+                eth_dst=setup["router_mac"],
+                eth_src=ptfadapter.dataplane.get_mac(0, 0),
+                ipv6_dst=dst_ip,
+                ipv6_src=src_ip,
+                icmp_type=8,
+                icmp_code=0,
+                ipv6_hlim=64,
+            )
+
+    def expected_mask_routed_packet(self, pkt, ip_version):
         """Generate the expected mask for a routed packet."""
         exp_pkt = pkt.copy()
-        exp_pkt["IP"].ttl -= 1
+
+        if ip_version == "ipv4":
+            exp_pkt["IP"].ttl -= 1
+        else:
+            exp_pkt["IPv6"].hlim -= 1
+
         exp_pkt = mask.Mask(exp_pkt)
         exp_pkt.set_do_not_care_scapy(packet.Ether, "dst")
         exp_pkt.set_do_not_care_scapy(packet.Ether, "src")
-        exp_pkt.set_do_not_care_scapy(packet.IP, "chksum")
+
+        if ip_version == "ipv4":
+            exp_pkt.set_do_not_care_scapy(packet.IP, "chksum")
+
         return exp_pkt
 
-    def test_unmatched_blocked(self, setup, direction, ptfadapter):
+    def test_unmatched_blocked(self, setup, direction, ptfadapter, ip_version):
         """Verify that unmatched packets are dropped."""
-        pkt = self.tcp_packet(setup, direction, ptfadapter)
-        self._verify_acl_traffic(setup, direction, ptfadapter, pkt, True)
+        pkt = self.tcp_packet(setup, direction, ptfadapter, ip_version)
+        self._verify_acl_traffic(setup, direction, ptfadapter, pkt, True, ip_version)
 
-    def test_source_ip_match_forwarded(self, setup, direction, ptfadapter, counters_sanity_check):
+    def test_source_ip_match_forwarded(self, setup, direction, ptfadapter, counters_sanity_check, ip_version):
         """Verify that we can match and forward a packet on source IP."""
-        pkt = self.tcp_packet(setup, direction, ptfadapter)
-        pkt["IP"].src = "20.0.0.2"
+        src_ip = "20.0.0.2" if ip_version == "ipv4" else "60c0:a800::6"
```

Comment:
How were these IPv4 IPv6 addresses chosen? You have a few combinations in this change where some pairs has matching last part, some doesn't.

Are they randomly chosen or have a rule?

Reply:
tbh they're pretty random, I can definitely go through and clean this up in a follow-on PR.